app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -316,6 +320,54 @@ protected function getRatesDebuggingData($rates) {
         return $rateDebuggingData;
     }
 
+    protected function isDiscountRoundingDeltaError($transaction, $quote) {
+        $boltDiscountAmount = $this->getBoltDiscountAmount($transaction);
+        $quoteDiscountAmount = $quote->getShippingAddress()->getDiscountAmount();
+
+        return $this->isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount);
+    }
+
+    protected function getBoltDiscountAmount($transaction) {
+        $boltDiscountAmount = 0;
+
+        if(isset($transaction->order->cart->discounts)) {
+            foreach($transaction->order->cart->discounts as $discount) {
+                if(isset($discount->amount->amount) && is_numeric($discount->amount->amount) && $discount->amount->amount > 0) {
+                    $boltDiscountAmount -= $discount->amount->amount / 100;
+                }
+            }
+        }
+
+        return $boltDiscountAmount;
+    }
+
+    protected function isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount) {
+        if($boltDiscountAmount != $quoteDiscountAmount) {
```

Comment:
are these float? What if `$boltDiscountAmount` is 10.00 and `$quoteDiscountAmount` is 10.000001 ?

Reply:
Both Bolt and Sales_Model_Service_Quote are rounded to two decimal places. In Magento collectTotals is always rounded just before returning the totals. If you think we should be more explicit we could round them to 2 decimal places here just to confirm. Thoughts?